homeassistant/components/sensor/wunderground.py

```diff
@@ -114,10 +123,15 @@ def __init__(self, rest, condition):
         """Initialize the sensor."""
         self.rest = rest
         self._condition = condition
+        self.entity_id = ENTITY_ID_FORMAT.format("pws_" + self._condition)
 
     @property
     def name(self):
         """Return the name of the sensor."""
+        if self._condition.startswith("forecast"):
+            period = self._condition[-1:]
+            return self.rest.data.get('forecast_title_' + period,
```

Comment:
If we get a proper human readable title, we should not overwrite the entity id so that the title will be used for the entity id too.